File: split/showcurcmdchr.pas
procedure showcurcmdchr;
begin
  begindiagnostic;
  printnl(123);
  if curlist.modefield <> shownmode then
  begin
    printmode(curlist.modefield);
    print(568);
    shownmode := curlist.modefield;
  end;
  printcmdchr(curcmd, curchr);
  printchar(125);
  enddiagnostic(false);
end;

File: cpp/showcurcmdchr.cpp
#include "showcurcmdchr.h"
#include "begindiagnostic.h"
#include "printnl.h"
#include "printmode.h"
#include "print.h"
#include "printcmdchr.h"
#include "printchar.h"
#include "enddiagnostic.h"

void showcurcmdchr(void)
{
	begindiagnostic();
	printnl('{');
	if (mode != shownmode)
	{
		printmode(mode);
		print(568); //: 
		shownmode = mode;
	}
	printcmdchr(curcmd, curchr);
	printchar('}');
	enddiagnostic(false);
}
